### PR TITLE
[Saas-9] reduce ormcache invalidation

### PR DESCRIPTION
**Pull request by rco-odoo**
Reduce the frequency of `ormcache` invalidation.
In particular, improve the conditions under which translations and users invalidate the cache.

### CODE REVIEW

File: openerp/addons/base/ir/ir_translation.py [hunk 1]
```diff
@@ -304,8 +304,25 @@ def _get_ids(self, cr, uid, name, tt, lang, ids):
...
+    CACHED_MODELS = {'ir.model.fields', 'ir.ui.view'}
+
+    def _modified_model(self, model_name):
+        """ Invalidate the ormcache if necessary, depending on ``model_name``.
+        This should be called when modifying translations of type 'model'.
+        """
+        if model_name in self.CACHED_MODELS:
+            self.clear_caches()
+
+    def _modified(self):
```
Comment (KangOl) [inline openerp/addons/base/ir/ir_translation.py]: missing `@api.multi` ?

File: openerp/addons/base/ir/ir_ui_view.py [hunk 2]
```diff
@@ -924,7 +924,7 @@ def postprocess_and_fields(self, cr, user, model, node, view_id, context=None):
 
     # apply ormcache_context decorator unless in dev mode...
     @tools.conditional(not config['dev_mode'],
-        tools.ormcache_context('uid', 'view_id',
+        tools.ormcache_context('frozenset(self.browse(cr, uid, uid).groups_id.ids)', 'view_id',
```
Comment (KangOl) [inline openerp/addons/base/ir/ir_ui_view.py]: Shouldn't be 
```python
frozenset(self.pool['res.users'].browse(cr, uid, uid).groups_id.ids)
```
or
```python
frozenset(self.env.user.groups_id.ids)
```
?

Comment (rco-odoo) [inline openerp/addons/base/ir/ir_ui_view.py]: Yes, you are absolutely right!
Thanks for pointing that out.

File: openerp/addons/base/ir/ir_translation.py (hunk 1 above)
Comment (rco-odoo) [inline openerp/addons/base/ir/ir_translation.py]: It is not used with the old API, so it is not strictly necessary.
But it's better with it than without it.

**Pull request merged**